internal/zbpack/zbpack.go

```diff
@@ -0,0 +1,115 @@
+// Package zbpack is internal package, contain the main logic of zbpack command-line interface.
+package zbpack
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"github.com/zeabur/zbpack/pkg/types"
+	"github.com/zeabur/zbpack/pkg/zeaburpack"
+)
+
+var (
+	// info option is used to analyze and print project information.
+	info bool
+	cmd  = &cobra.Command{
+		Use:   "zbpack",
+		Short: "Zbpack is a tool to help you analyze your project and build Docker image in one click.",
+		Long: "Zbpack is a powerful tool that not only analyzes your project for dependencies and requirements, " +
+			"but also builds Docker images in one click, greatly simplifying your workflow.",
+		Args: func(cmd *cobra.Command, args []string) error {
+			if err := cobra.MinimumNArgs(1)(cmd, args); err != nil {
+				return fmt.Errorf("zbpack requires a directory to analyse or build")
+			}
+			return nil
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return run(args)
+		},
+	}
+)
+
+func init() {
+	cmd.PersistentFlags().BoolVarP(&info, "info", "i", false, "only print project information")
+	cmd.SetUsageTemplate(usageTemplate)
+}
+
+// Execute is used to execute zbpack command-line interface.
+func Execute() error {
+	return cmd.Execute()
+}
+
+// run is command-line entrypoint.
+func run(args []string) error {
+	path := args[0]
+
+	switch {
+	case info:
+		return plan(path)
+	default:
+		return build(path)
+	}
+}
+
+// build is used to build Docker image and show build plan.
+func build(path string) error {
+	if err := CheckDockerDaemonStatus(); err != nil {
+		return err
+	}
+
+	if strings.HasPrefix(path, "https://") || strings.HasPrefix(path, "http://") {
+		return fmt.Errorf("zbpack does not support building from online repositories yet")
+	}
```

Comment:
hmmmm i think it is possible by cloning the repository to `/tmp` and run a build on that directory; however, it should be written in another PR. At this moment, I recommend you to add a `// TODO` mark.